### PR TITLE
fix: duplicated comments

### LINKED ISSUE #399: 🐛 Bug: comments are duplicated on prettier run
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Comments should survive "as they are"

```ts
//pre comment
if (true)
    //pre branch comment
    console.log('hurray')
//post comment
```
should get

```ts
//pre comment
if (true) {
    //pre branch comment
    console.log('hurray')
}
//post comment
```

### Actual

Formatting results in:

```ts
//pre comment

//pre comment
if (true) {
    //pre branch comment
    console.log('hurray');
}
//post comment
//post comment
```

### Additional Info

ref #179

### PR DESCRIPTION
**Pull request by RobinHeidenis**
## PR Checklist

- [x] Addresses an existing open issue: fixes #399 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

When adding braces to (if) statements that have comments above or below them they would be reprinted, resulting in duplicated comments. This PR aims to fix that by deleting the `leadingComments` and `trailingComments` properties from the AST nodes parsed by the plugin.

Removing these can cause extra newlines being printed (in place of the comments), so they are being filtered out by `trim()`ing the generated code.

I've ran this modified version of the plugin on my company's internal project, and I haven't seen anything weird or wrong being outputted. I've also added two tests to verify the behavior. 

I'm not sure if deleting the `leadingComments` and `trailingComments` is the best way to do this, but it does seem to work properly. Perhaps someone else could verify these changes, as my company's codebase has mostly if statements that are being rewritten by this plugin, and not any of the other examples in the tests like while and for statements.

🚀

### CODE REVIEW

**Comment by RobinHeidenis** [comment]
Small note: I couldn't get ESLint to work locally, as it's installing v9 and the eslintrc is still in the old format. Installing ESLint v8 manually results in a circular dependency in the config with the vitest plugin. Hopefully it works in the pipelines

File: src/traverseAndModifyAst.ts [hunk 1]
```diff
@@ -15,6 +15,14 @@ export function traverseAndModifyAst(ast: Node) {
 			modifyNodeIfMissingBrackets(node) &&
 			nodeNotAlreadySeen(node, seenNodes)
 		) {
+			// Remove comments from the original node to avoid duplication
+			if (node.leadingComments) {
+				delete node.leadingComments;
+			}
+			if (node.trailingComments) {
+				delete node.trailingComments;
+			}
```
Comment (JoshuaKGoldberg) [inline src/traverseAndModifyAst.ts:24]: [Question] Any reason for the `if`s? No tests fail if I simplify them out:

```suggestion
			delete node.leadingComments;
			delete node.trailingComments;
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Looks great to me, nice new test coverage! ✨ 

Just one question/suggestion on simplifying a bit, then I think we're good to go.

File: src/preprocess.test.ts
Comment (JoshuaKGoldberg) [inline src/preprocess.test.ts:1]: > Perhaps someone else could verify these changes, as my company's codebase has mostly if statements that are being rewritten by this plugin, and not any of the other examples in the tests like while and for statements.

👍 yeah agreement from me, I can't find any issues. If anything pops up in the future then great, it should be represented here as a test case.

**Comment by JoshuaKGoldberg** [comment]
If you merge from the latest `main` then the unrelated Lint and Prettier failures should go away. Sorry for the noise. 🙂

File: src/traverseAndModifyAst.ts (hunk 1 above)
Comment (RobinHeidenis) [reply src/traverseAndModifyAst.ts]: Good catch :D I forgot delete doesn't throw if the property doesn't exist on the object, so the if statements are unnecessary. I've removed them

**Review by JoshuaKGoldberg — APPROVED** [review]
🎇

**Pull request merged**